src/sdl2/render.rs

```diff
@@ -380,13 +519,18 @@ impl Error for TextureValueError {
 }
 
 /// Texture-creating methods for the renderer
-impl<'a> Renderer<'a> {
+impl<T> TextureCreator<T> {
```

Comment:
I don't think we should prevent Texture creation from the Renderer itself. Unless you did that on purpose (then i'd like to hear your explanation), you should restore the texture creation method to Canvas.

Currently the canvas doesn't have a way to create a texture without creating a TextureCreator, but there are still cases where you will want to create textures without creating a TextureCreator.